packages/server/src/utils/index.ts

```diff
@@ -406,6 +409,8 @@ export const buildFlow = async (
         for (let i = 0; i < neighbourNodeIds.length; i += 1) {
             const neighNodeId = neighbourNodeIds[i]
             if (ignoreNodeIds.includes(neighNodeId)) continue
+            if (initializedNodes.has(neighNodeId)) continue
+            if (nonDirectedGraph[neighNodeId].some((dependId) => !initializedNodes.has(dependId))) continue
```

Comment:
I dont quite understand this line, why its it necessary? and can you walk me through a simple example

Reply:
please look at this graph

```mermaid
  graph LR;
      Configration --> Initialization
      Initialization -- Reuse results--> EndNode
      Initialization --> A
      A --> B
      B --> C
      C --> EndNode
```

EndNode will be calculated via path from Initializaton to EndNode, at this time, C is unintialized.

So it would better to wait until from path C to EndNode.

Reply:
Okay, for future reference:

When we arrive at `Initialization`, we have 2 neighbours: `A` and `EndNode`.

For `EndNode`:
- `reverseGraph[EndNode] = [Initialization, C]`
- `initializedNodes = [Initialization]`
- Since C is not part of `initializedNodes`, we skip processing `EndNode`

For `A`:
- `reverseGraph[A] = [Initialization]`
- `initializedNodes = [Initialization]`
- Continue processing A